docs/piwiks-extensibility-points.md

```diff
@@ -5,189 +5,8 @@ category: CoreDevelop
-{
-    return array(
-        'API.getSegmentsMetadata' => array(
-            'after'    => true,
-            'function' => 'getSegmentsMetadata',
-        )
-    );
-}
-```
-
-*Note: You can also use these options when calling [`Piwik::addAction()`](/api-reference/Piwik/Piwik#addaction).*
-
-### Posting events
-
-Plugins can post events themselves if they want to provide extension points to other plugins. For example, the [ScheduledReports](https://github.com/piwik/piwik/tree/master/plugins/ScheduledReports) plugin post events to allow plugins to define new transport mediums and report formats.
-
-To post an event, call [`Piwik::postEvent()`](/api-reference/Piwik/Piwik#postevent) using the name of your event:
-
-```php
-Piwik::postEvent('MyPluginOrClass.MyEvent', array($myFirstArg, &$myRefArg));
-```
-
-#### Event naming conventions
-
-Event names should follow this format: `$scopeName.$shortEventDescription` where `$scopeName` is the name of the plugin or class that is posting the event and where `$shortEventDescription` is a short description of the event's purpose, for example, **getAvailableViewDataTables**.
-
-#### Posting events in templates
-
-You can post events within Twig templates by using the `postEvent()` function:
-
-```twig
-{{ postEvent('MyPlugin.MyEventInATemplate') }}
-```
-
-This function will pass a string by reference to all event handlers and then insert the string into the template. Event handlers can modify the string, inserting HTML into templates in other plugins:
-
-```php
-class MyOtherPlugin extends \Piwik\Plugin
-{
-    public function getListHooksRegistered()
-    {
-        return array(
-            'MyPlugin.MyEventInATemplate' => 'handleMyEventInATemplate',
-        );
-    }
-
-    public function handleMyEventInATemplate(&$outString)
-    {
-        $outString .= '<h1>This text was injected!</h1>';
-    }
-}
-```
-
-When posting events, the templates can pass extra parameters:
-
-```twig
-{{ postEvent('MyPlugin.MyEventInATemplate', usefulVariable1, usefulVariable2) }}
-```
-
-Event handlers can read these optional values as follows:
-
-```php
-public function handleMyEventInATemplate(&$outString, $usefulVariable1, $usefulVariable2)
-{
-    $outString .= '<h1>This text was injected!</h1>';
-    $outString .= $usefulVariable . ' - ' . $usefulVariable2;
-}
-```
-
-## Special plugin classes
-
-Plugins can define certain special classes in order to extend Piwik. These classes must be placed in the root namespace of the plugin. Piwik will automatically detect, instantiate and use them.
-
-### API and Controller
-
-Plugins can define an **API** class (that extends [Piwik\Plugin\API](/api-reference/Piwik/Plugin/API)) to add more methods to the [Reporting API](/guides/piwiks-reporting-api). They can also define a **Controller** class to handle HTTP requests that are sent by Piwik's UI.
-
-*Learn more about these classes in our [Controllers](/guides/controllers) or [APIs](/guides/apis) guides.*
-
-### Archiver
```

Comment:
Archiver is way too complex and needs to be refactored and documented in a separate guide